support/agenda-utils.js

```diff
@@ -19,15 +17,19 @@ async function getDataForAgenda(meetingUuid, agendaKindUuid) {
   const agendapoints = await AgendaPoint.findAll({meetingUuid: meetingUuid});
   if (agendaKindUuid) {
     const agendaType = await Concept.find(agendaKindUuid);
-    for (const agendapoint of agendapoints) {
-      if (! agendapoint.type) {
-        agendapoint.type = agendaType.uri;
-        agendapoint.typeName = agendaType.label;
-      }
-    }
+    ensureAgendapointType(agendapoints, agendaType);
   }
   return { meeting, agendapoints };
 }
+
+export function ensureAgendapointType(agendapoints, type) {
```

Comment:
I don't understand the name of this function, why ensure instead of something like assign?

Reply:
cause we don't overwrite if it already has one. Assign would imply overwriting, and I think we use ensure in other places to signify that we only create/add information if it's not already there

Reply:
But when it won't have one? I think we are fetching then without one and then adding them

Reply:
We fetch it optionally, so indeed we only set if it's not set yet. e.g. if you haven't published any agenda yet.
e.g. :
- you publish the planned agenda: all agendapoints get the type planned
- you add some agendapoints
- you publish the additional agenda this contains the earlier agendapoints with type planned and the new ones with type "additional"

Reply:
Ohhh indeed, then it makes sense thanks for explaining :)